internal/isoeditor/rhcos.go

```diff
@@ -541,3 +546,49 @@ func ParseOffsetInfo(headerBytes []byte) (*OffsetInfo, error) {
 	}
 	return offsetInfo, nil
 }
+
+func EmbedIgnition(inputISOPath, outputISOPath, ignitionConfig string) error {
+	coreosIgnitionHeader := make([]byte, ignitionHeaderSize)
+
+	inputISO, err := os.Open(inputISOPath)
+	if err != nil {
+		return err
+	}
+	defer inputISO.Close()
+
+	if _, err = inputISO.ReadAt(coreosIgnitionHeader, isoSystemAreaSize-ignitionHeaderSize); err != nil {
```

Comment:
maybe worth just adding a comment that we're reading 24 bytes from the end of system area.